### PR TITLE
Merge recent 4.x changes

### PR DESCRIPTION
**Pull request by kallaballa**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: doc/footer.html [hunk 1]
```diff
@@ -10,17 +10,17 @@
...
 <!--BEGIN !GENERATE_TREEVIEW-->
 <hr class="footer"/><address class="footer"><small>
 $generatedby &#160;<a href="http://www.doxygen.org/index.html">
 <img class="footer" src="$relpath^doxygen.png" alt="doxygen"/>
 </a> $doxygenversion
 </small></address>
 <!--END !GENERATE_TREEVIEW-->
-<script type="text/javascript">
-//<![CDATA[
-addTutorialsButtons();
```
Comment (opencv-alalek) [inline doc/footer.html:22]: Why?

File: modules/core/src/ocl.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*M///////////////////////////////////////////////////////////////////////////////////////
+    /*M///////////////////////////////////////////////////////////////////////////////////////
```
Comment (opencv-alalek) [inline modules/core/src/ocl.cpp:1]: there are too much random unnecessary invalid changes.

**Review by opencv-alalek — COMMENTED** [review]
Please specify clear purpose of this PR.

File: modules/core/src/ocl.cpp (hunk 2 above)
Comment (kallaballa) [reply modules/core/src/ocl.cpp:1]: Please disregard. Wrong direction

Comment (kallaballa) [reply modules/core/src/ocl.cpp:1]: Didn't realize i actually created a PR

**Pull request closed without merging**